database/skylink.go

```diff
@@ -91,29 +83,21 @@ func (db *DB) FindSkylink(ctx context.Context, skylink string) (Skylink, error)
 
 // MarkPinned marks a skylink as pinned (or no longer unpinned), meaning
 // that Pinner should make sure it's pinned by the minimum number of servers.
-func (db *DB) MarkPinned(ctx context.Context, skylink string) error {
-	sl, err := CanonicalSkylink(skylink)
-	if err != nil {
-		return ErrInvalidSkylink
-	}
-	filter := bson.M{"skylink": sl}
+func (db *DB) MarkPinned(ctx context.Context, skylink skymodules.Skylink) error {
```

Comment:
`unpin` is not a state, it's an instruction. That's why it's a verb and not an adjective. It tells servers that they should actively stop pinning this skylink because no user is paying for it anymore.

From the docstring of `Skylink`:
```
	// The Unpin field instructs all servers, currently pinning this skylink,
	// that there are no users pinning and the servers should unpin it as well.
```

I simplified that a bit to
```
		// Unpin informs all servers that there are no users pinning and they
		// should stop pinning it as well.
```

Reply:
`update := bson.M{"$set": bson.M{"unpin": false}}` this is what I meant, the field is called `unpin`, my suggestion was to maybe call it `pinned` because then you avoid a double negative (MarkPinned -> set unpin:false becomes set pinned:true) and Unpin becomes set pinned:false which is generally more readable

I was just wondering though, it might very well be that that's not the case in this particular case, just usually if I see a double negative and in general "negative" properties or fields, it's often better to flip it to the positive side and avoid double negatives.

Reply:
I agree with PJ on the naming. I think flipping it to `pinned` would be more readable.

Reply:
https://github.com/SkynetLabs/pinner/issues/10